### PR TITLE
OSDOCS#8773: Agent configuration reference assembly

### PR DESCRIPTION
**Pull request by skopacz1**
[OSDOCS-8773](https://issues.redhat.com/browse/OSDOCS-8773)

Version(s):
4.14+

(content should be backported in later doc effort, I believe the main difference would be which platforms are supported)

This PR adds a new assembly containing all the available configuration parameters for the `install-config.yaml` and `agent-config.yaml` files used during an Agent-based installation.

QE review:
- [x] QE has approved this change.

Preview: [Installation configuration parameters for the Agent-based Installer](https://69292--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_with_agent_based_installer/installation-config-parameters-agent)

### CODE REVIEW

File: modules/agent-configuration-parameters.adoc [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+|Host configuration.
+The number of hosts defined must not exceed the total number of hosts defined in the `install-config.yaml` file, which is the sum of the values of the `compute.replicas` and `controlPlane.replicas` parameters.
+|An array of host configuration objects.
+
+|hosts:
+  hostname:
+|Host name.
+Overrides the hostname obtained from either the Dynamic Host Configuration Protocol (DHCP) or a reverse DNS lookup.
+Each host must have a unique hostname supplied by one of these methods.
+|(needs info)
```
Comment (bfournie) [inline modules/agent-configuration-parameters.adoc:84]: This seems reasonable, could also note that the hostname supplied here is optional.

File: modules/agent-configuration-parameters.adoc [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+|hosts:
+  interfaces:
+|(needs info)
+|(needs info)
```
Comment (bfournie) [inline modules/agent-configuration-parameters.adoc:89]: The interface table provides a name and mac address mapping for the interfaces on the host. The settings must match the configuration in the NetworkConfig section. This table is optional, but if the NetworkConfig section is included it must also e provided.

File: modules/agent-configuration-parameters.adoc [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    macAddress:
+|(needs info)
+|(needs info)
+
+|hosts:
+  rootDeviceHints:
+|Enables provisioning of the {op-system-first} image to a particular device.
+It examines the devices in the order it discovers them, and compares the discovered values with the hint values.
+It uses the first discovered device that matches the hint value.
+|(needs info)
```
Comment (bfournie) [inline modules/agent-configuration-parameters.adoc:108]: Maybe in addition state that this is the device that the OS will be written during installation.

File: modules/agent-configuration-parameters.adoc [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+|hosts:
+  rootDeviceHints:
+    deviceName:
+|(needs info)
+|(needs info)
+
+|hosts:
+  networkConfig:
```
Comment (bfournie) [inline modules/agent-configuration-parameters.adoc:117]: The networkConfig field is the host network definition. It must match the Network Management API defined in https://nmstate.io/.

File: modules/agent-configuration-parameters.adoc [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+|(needs info)
+|(needs info)
+
```
Comment (bfournie) [inline modules/agent-configuration-parameters.adoc:120]: I think all of the fields below can be removed. The NetworkConfig must match the fields in https://nmstate.io/, its not necessary to redefine them here.

File: installing/installing_with_agent_based_installer/installation-config-parameters-agent.adoc [hunk 6]
```diff
@@ -0,0 +1,14 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="installation-config-parameters-agent"]
+= Installation configuration parameters for the Agent-based Installer
+include::_attributes/common-attributes.adoc[]
+:context: installation-config-parameters-agent
+
+toc::[]
+
+Before you deploy an {product-title} cluster using the Agent-based Installer, you provide parameters to customize your cluster and the platform that hosts it.
+When you create the `install-config.yaml` and `agent-config.yaml` files, you provide values for the required parameters through the command line. You can then modify the `install-config.yaml` and `agent-config.yaml` files to customize your cluster further.
```
[the remark this answers is not in the document]
Comment (skopacz1) [reply installing/installing_with_agent_based_installer/installation-config-parameters-agent.adoc:10]: I'm not sure either, I took this wording from similar reference pages in other installer sections and modified it to include agent-config. Would this phrasing seem better to you?
```suggestion
When you create the `install-config.yaml` and `agent-config.yaml` files, you must provide values for the required parameters, and you can use the optional parameters to customize your cluster further.
```

File: modules/agent-configuration-parameters.adoc [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+|apiVersion:
+|The API version for the `agent-config.yaml` content.
+The current version is v1.
+The installation program may also support older API versions.
+|String
+
+|kind:
+|(needs info)
+|(needs info)
```
[the remark this answers is not in the document]
Comment (skopacz1) [reply modules/agent-configuration-parameters.adoc:35]: Sounds good, I can delete this parameter

File: modules/agent-configuration-parameters.adoc [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+|additionalNTPSources:
+|(needs info)
+|(needs info)
+
+|hosts:
+|Host configuration.
+The number of hosts defined must not exceed the total number of hosts defined in the `install-config.yaml` file, which is the sum of the values of the `compute.replicas` and `controlPlane.replicas` parameters.
+|An array of host configuration objects.
+
+|hosts:
```
[the remark this answers is not in the document]
Comment (skopacz1) [reply modules/agent-configuration-parameters.adoc:79]: This is the new format that the install config reference is being represented as. Previously the format would be `hosts.interfaces.macAddress`, however this would make the table cells really wide and practically unreadable in some cases.

File: modules/agent-configuration-parameters.adoc (hunk 5 above)
Comment (skopacz1) [reply modules/agent-configuration-parameters.adoc:120]: Sounds good, I'll delete these remaining NetworkConfig parameters

File: modules/agent-configuration-parameters.adoc [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+.Required parameters
+[cols=".^2l,.^3,.^5a",options="header"]
+|====
+|Parameter|Description|Values
+
+|apiVersion:
+|The API version for the `agent-config.yaml` content.
+The current version is v1.
+The installation program may also support older API versions.
```
Comment (bfournie) [inline modules/agent-configuration-parameters.adoc:30]: The current - and the only version defined so far - is` v1beta1`

File: modules/agent-configuration-parameters.adoc [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+|The MAC address of an interface on the host.
+|A MAC address such as the following example: `00-B0-D0-63-C2-26`
+
+|hosts:
+  rootDeviceHints:
+|Enables provisioning of the {op-system-first} image to a particular device.
+It examines the devices in the order it discovers them, and compares the discovered values with the hint values.
+It uses the first discovered device that matches the hint value.
+This is the device that the operating system will be written on during installation.
+|(needs accepted/valid values)
```
Comment (bfournie) [inline modules/agent-configuration-parameters.adoc:107]: I don't think need to list accepted/valid values here, it should just refer to other section in the docs that describes rootDeviceHints e.g. https://docs.openshift.com/container-platform/4.13/installing/installing_bare_metal_ipi/ipi-install-installation-workflow.html#root-device-hints_ipi-install-installation-workflow

File: modules/agent-configuration-parameters.adoc [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+  interfaces:
+|Provides a table of the name and MAC address mappings for the interfaces on the host.
+If a `NetworkConfig` section is provided in the `agent-config.yaml`, this table must be included and the values must match the mappings provided in the `NetworkConfig` section.
+|An array of host configuration objects.
+
+|hosts:
+  interfaces:
+    name:
+|The name of an interface on the host.
+|(needs accepted/valid values)
```
Comment (bfournie) [inline modules/agent-configuration-parameters.adoc:93]: Interface names aren't really checked for validity so don't need to list valid values here.

Comment (skopacz1) [reply modules/agent-configuration-parameters.adoc:93]: If a string of characters is accepted in general, I can just put "String" here. Let me know if that's not accurate.

Comment (bfournie) [reply modules/agent-configuration-parameters.adoc:93]: Yes that's fine

File: modules/agent-configuration-parameters.adoc (hunk 10 above)
Comment (skopacz1) [reply modules/agent-configuration-parameters.adoc:107]: I still think there might be a need to define the type of data that should be used with this parameter (and all others, even if we don't want to specify which specific values are accepted). Would this be an accurate entry to include here?

`A dictionary of key-value pairs. For more information, see Root device hints.`

With a link to the section you provide in your comment

Comment (bfournie) [reply modules/agent-configuration-parameters.adoc:107]: Sure

**Comment by skopacz1** [comment]
@bfournie and other Agent devs, I think I have enough information to move this PR on to QE review, thanks for all the help! If you'd like to make more comments/feedback while I'm waiting on or implementing QE review, feel free to keep on doing so.

**Comment by skopacz1** [comment]
@mhanss could you PTAL when you have a chance! Thank you!

File: modules/agent-configuration-parameters.adoc [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+The installation program may also support older API versions.
+|String
+
+|metadata:
+|Kubernetes resource `ObjectMeta`, from which only the `name` parameter is consumed.
+|Object
+
+|metadata:
+  name:
+|The name of the cluster.
```
Comment (mhanss) [inline modules/agent-configuration-parameters.adoc:39]: The parameter `metadata.name` is optional; however, when included in the agent-config file, it takes precedence over the value specified in the install-config file.

Comment (mhanss) [reply modules/agent-configuration-parameters.adoc:39]: If included in the agent-config file, the metadata.name parameter neither takes precedence nor serves as a required or optional setting; rather, it is simply ignored, and the value specified in the install-config file is used.
@bfournie please confirm.

**Comment by mhanss** [comment]
@skopacz1 I would suggest omitting unsupported platforms from the install-config in case of agent-based installation.

File: modules/agent-configuration-parameters.adoc (hunk 12 above)
Comment (bfournie) [reply modules/agent-configuration-parameters.adoc:39]: The metadata name and namespace are required. From the agent-config.yaml defintion: https://github.com/openshift/installer/blob/master/pkg/types/agent/agent_config_type.go#L19-L20

However they are not validated or used. As Manoj indicated, the fields are ignored

Comment (skopacz1) [reply modules/agent-configuration-parameters.adoc:39]: I'll add a sentence to this entry stating "The value entered in the `agent-config.yaml` file is ignored, and the instead the value specified in the `install-config.yaml` file is used."

Let me know if that sounds right, or if I need to add to/change that sentence.

File: modules/installation-configuration-parameters.adoc
Comment (skopacz1) [inline modules/installation-configuration-parameters.adoc:1]: @mhanss let me know if these were the values you had in mind for me modifying to omit unsupported platform: `platform`, `compute.platform`, and `controlPlane.platform`

**Comment by skopacz1** [comment]
@mhanss thanks for the feedback, let me know if I'm missing anything or need to further refine this PR.

File: modules/agent-configuration-parameters.adoc (hunk 12 above)
Comment (bfournie) [reply modules/agent-configuration-parameters.adoc:39]: That sounds right, although I did discover that if the install-config.yaml is ALSO not supplied that a default cluster name of `agent-cluster` will be used, see https://github.com/openshift/installer/blob/master/pkg/asset/agent/installconfig.go#L253-L258

Comment (skopacz1) [reply modules/agent-configuration-parameters.adoc:39]: @bfournie just out of curiosity, how is this value a required parameter in both the agent-config and install-config files if the installation proceeds without them specified? I'm not familiar with OCP configuration, does "required" not mean "necessary for the installation and thus the installer will terminate with an error without this value"?

Comment (mhanss) [reply modules/agent-configuration-parameters.adoc:39]: @bfournie If the value is not specified in the install-config, the following error message will be generated: "FATAL failed to fetch Agent Manifests: failed to load asset "Install Config": invalid install-config configuration: metadata.name: Invalid value: "": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character"

File: modules/installation-configuration-parameters.adoc
Comment (mhanss) [reply modules/installation-configuration-parameters.adoc:1]: Yes, however, it is also essential to include 'external' as a platform value to consider the support for OCI. For example:
```
platform:
  external:
    platformName: oci
    cloudControllerManager: External

Comment (skopacz1) [reply modules/installation-configuration-parameters.adoc:1]: That's being added for 4.15, right? If so, can I add external in a later PR before the 4.15 GA? I'm hoping to merge this PR into 4.12+.

File: modules/agent-configuration-parameters.adoc (hunk 12 above)
Comment (bfournie) [reply modules/agent-configuration-parameters.adoc:39]: For the install-config.yaml, if it's provided the metadata.name must be included. However install-config.yaml is optional for the agent installer (when using just the ZTP manifests). In that case the cluster name will be set to `agent-cluster`

File: modules/installation-configuration-parameters.adoc
Comment (skopacz1) [reply modules/installation-configuration-parameters.adoc:1]: Wait, I see in https://issues.redhat.com/browse/AGENT-656 that platform external for OCI was added for 4.14. I can add external to this PR and make this PR for 4.14+, and then backport this content to 4.12 and 4.13 later.

File: modules/agent-configuration-parameters.adoc (hunk 12 above)
Comment (skopacz1) [reply modules/agent-configuration-parameters.adoc:39]: Proposed sentence to add to `metadata.name` descriptions for both agent-config and install-config: 

"When you do not provide `metadata.name` through either the `install-config.yaml` or `agent-config.yaml` files, for example when you use only ZTP manifests, the cluster name will be set to `agent-cluster`."

File: modules/installation-configuration-parameters.adoc [hunk 13]
```diff
@@ -491,7 +516,12 @@ endif::vsphere[]
 ifdef::ibm-power-vs[]
 Example usage, `compute.platform.powervs.sysType`.
 endif::ibm-power-vs[]
+ifndef::agent[]
 |`alibabacloud`, `aws`, `azure`, `gcp`, `ibmcloud`, `nutanix`, `openstack`, `powervs`, `vsphere`, or `{}`
+endif::agent[]
+ifdef::agent[]
+|`vsphere`
```
Comment (skopacz1) [inline modules/installation-configuration-parameters.adoc:523]: @mhanss do I need to add `{}` and/or `external` to this parameter?

File: modules/installation-configuration-parameters.adoc [hunk 14]
```diff
@@ -572,7 +602,12 @@ endif::vsphere[]
 ifdef::ibm-power-vs[]
 Example usage, `controlPlane.platform.powervs.processors`.
 endif::ibm-power-vs[]
+ifndef::agent[]
 |`alibabacloud`, `aws`, `azure`, `gcp`, `ibmcloud`, `nutanix`, `openstack`, `powervs`, `vsphere`, or `{}`
+endif::agent[]
+ifdef::agent[]
+|`vsphere`
```
Comment (skopacz1) [inline modules/installation-configuration-parameters.adoc:609]: @mhanss do I need to add `{}` and/or `external` to this parameter?

File: modules/agent-configuration-parameters.adoc [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+[cols=".^2l,.^4,.^4a",options="header"]
+|====
+|Parameter|Description|Values
+
+|rendezvousIP:
+|The IP address of the node that performs the bootstrapping process as well as running the `assisted-service` component.
+You must provide the rendezvous IP address when you do not specify at least one host's IP address in the `networkConfig` parameter.
+If this address is not provided, one IP address is selected from the provided hosts' `networkConfig`.
+|IPv4 or IPv6 address.
+IPv6 is supported only on bare metal platforms.
```
Comment (mhanss) [inline modules/agent-configuration-parameters.adoc:62]: It is also supported on vSphere platforms.

File: modules/installation-configuration-parameters.adoc
Comment (mhanss) [inline modules/installation-configuration-parameters.adoc:1]: We need to update the network configuration and some other values in the install-config. You can refer to this documentation for guidance on network configuration. https://docs.openshift.com/container-platform/4.14/installing/installing_bare_metal/installation-config-parameters-bare-metal.html#installation-configuration-parameters-network_installation-config-parameters-bare-metal.

- The architecture values for Compute and controlPlane should be either amd64 or arm64.
- In the case of SNO, the controlPlane replicas value can be set to 1.
- IPv6 is supported for both bare-metal and vSphere platforms. Therefore, the statement indicating that IPv6 is only supported for bare-metal can be removed.

Comment (skopacz1) [reply modules/installation-configuration-parameters.adoc:1]: I've taken the following actions in response to this comment:

1.  Made the network config section in Agent look the same as the network config in the bare metal page you linked to, which includes the content that mentions both IPv4 and IPv6.
2. Modified the `compute.architecture` and `controlPlane.architecture` parameters to include both of those architectures in the Agent doc.
3. Modified `controlPlane.replicas` to say that `1` is a valid value for SNO. This change will appear in all the install sections that have this reference info, not just Agent.
4. Removed content in the Agent page that states either "only IPv4 is supported" or "IPv6 is supported only for bare metal", these changes are applied to both the `install-config` and `agent-config` sections.

File: modules/installation-configuration-parameters.adoc (hunk 13 above)
Comment (pawanpinjarkar) [reply modules/installation-configuration-parameters.adoc]: @skopacz1 You can set `controlPlane.Platform` and `compute.Platform` to `{}` for agent installer like below:
```
            "controlPlane": {
                "name": "master",
                "replicas": 1,
                "platform": {},
                "hyperthreading": "Enabled",
                "architecture": "amd64"
            },
            "compute": [
                {
                    "name": "worker",
                    "replicas": 0,
                    "platform": {},
                    "hyperthreading": "Enabled",
                    "architecture": "amd64"
                }
            ],
```
I don't think you need to specify other possible platform values in there. The possible values could be ( no external here),
```
"controlPlane": {
                "name": "master",
                "replicas": 1,
                "platform": {
                    "aws": {
                        "zones": ["us-east", "us-west"],
                        "type": "test"
                    }
                },
                "hyperthreading": "Enabled",
                "architecture": "amd64"
            },
```
Here is the schema definition for `platform` section for controlPlane and compute https://github.com/openshift/installer/blob/master/pkg/types/machinepools.go#L64

Each of the underlying platform schema definition is here https://github.com/openshift/installer/blob/master/pkg/types/machinepools.go#L84 and any specific ones are here https://github.com/openshift/installer/blob/master/pkg/types/aws/machinepool.go#L5



Long story short, for agent installer, you can just set `controlPlane.Platform` and `compute.Platform`  to `{}`

**Comment by mhanss** [comment]
thanks @skopacz1, LGTM.

File: modules/installation-configuration-parameters.adoc (hunk 13 above)
Comment (skopacz1) [reply modules/installation-configuration-parameters.adoc]: Thanks for the detailed response, Pawan!

File: modules/agent-configuration-parameters.adoc [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+
+.Required parameters
+[cols=".^2l,.^3,.^5a",options="header"]
+|====
+|Parameter|Description|Values
+
+|apiVersion:
+|The API version for the `agent-config.yaml` content.
+The current version is `v1beta1`.
+The installation program may also support older API versions.
```
Comment (kcarmichael08) [inline modules/agent-configuration-parameters.adoc:30]:
```suggestion
The installation program might also support older API versions.
```
See [IBM Style](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#m)

File: installing/installing_with_agent_based_installer/installation-config-parameters-agent.adoc [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+When you create the `install-config.yaml` and `agent-config.yaml` files, you must provide values for the required parameters, and you can use the optional parameters to customize your cluster further.
+
+include::modules/installation-configuration-parameters.adoc[leveloffset=+1]
+
+include::modules/agent-configuration-parameters.adoc[leveloffset=+1]
+
+[role="_additional-resources"]
+.Additional resources
+* xref:../../installing/installing_with_agent_based_installer/prepare-pxe-assets-agent.adoc#prepare-pxe-assets-agent[Preparing PXE assets for {product-title}].
+* xref:../../installing/installing_bare_metal_ipi/ipi-install-installation-workflow.adoc#root-device-hints_ipi-install-installation-workflow[Root device hints].
```
Comment (kcarmichael08) [inline installing/installing_with_agent_based_installer/installation-config-parameters-agent.adoc:19]:
```suggestion
* xref:../../installing/installing_bare_metal_ipi/ipi-install-installation-workflow.adoc#root-device-hints_ipi-install-installation-workflow[Root device hints]
```

Comment (kcarmichael08) [inline installing/installing_with_agent_based_installer/installation-config-parameters-agent.adoc:19]:
```suggestion
* xref:../../installing/installing_bare_metal_ipi/ipi-install-installation-workflow.adoc#root-device-hints_ipi-install-installation-workflow[Root device hints]
```

File: modules/agent-configuration-parameters.adoc [hunk 18]
```diff
@@ -0,0 +1,123 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_with_agent_based_installer/installation-config-parameters-agent.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="agent-configuration-parameters_{context}"]
+= Available Agent configuration parameters
```
Comment (kcarmichael08) [inline modules/agent-configuration-parameters.adoc:7]: missing blank line after header?

File: modules/agent-configuration-parameters.adoc [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+
+|metadata:
+|Kubernetes resource `ObjectMeta`, from which only the `name` parameter is consumed.
+|Object
+
+|metadata:
+  name:
+|The name of the cluster.
+DNS records for the cluster are all subdomains of `{{.metadata.name}}.{{.baseDomain}}`.
+The value entered in the `agent-config.yaml` file is ignored, and the instead the value specified in the `install-config.yaml` file is used.
```
Comment (kcarmichael08) [inline modules/agent-configuration-parameters.adoc:41]:
```suggestion
The value entered in the `agent-config.yaml` file is ignored, and instead the value specified in the `install-config.yaml` file is used.
```

File: modules/agent-configuration-parameters.adoc [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+When you do not provide `metadata.name` through either the `install-config.yaml` or `agent-config.yaml` files, for example when you use only ZTP manifests, the cluster name will be set to `agent-cluster`.
```
Comment (kcarmichael08) [inline modules/agent-configuration-parameters.adoc:42]:
```suggestion
When you do not provide `metadata.name` through either the `install-config.yaml` or `agent-config.yaml` files, for example when you use only ZTP manifests, the cluster name is set to `agent-cluster`.
```

File: modules/agent-configuration-parameters.adoc [hunk 21]
```diff
@@ -0,0 +1,123 @@
...
+    macAddress:
+|The MAC address of an interface on the host.
+|A MAC address such as the following example: `00-B0-D0-63-C2-26`
+
+|hosts:
+  rootDeviceHints:
+|Enables provisioning of the {op-system-first} image to a particular device.
+It examines the devices in the order it discovers them, and compares the discovered values with the hint values.
+It uses the first discovered device that matches the hint value.
+This is the device that the operating system will be written on during installation.
```
Comment (kcarmichael08) [inline modules/agent-configuration-parameters.adoc:108]:
```suggestion
This is the device that the operating system is written on during installation.
```

File: modules/agent-configuration-parameters.adoc [hunk 22]
```diff
@@ -0,0 +1,123 @@
...
+|A dictionary of key-value pairs.
+For more information, see "Root device hints" in the "Setting up the environment for an OpenShift installation" page.
+
+|hosts:
+  rootDeviceHints:
+    deviceName:
+|The name of the device the {op-system} image will be provisioned to.
```
Comment (kcarmichael08) [inline modules/agent-configuration-parameters.adoc:115]:
```suggestion
|The name of the device the {op-system} image is provisioned to.
```

File: modules/installation-configuration-parameters.adoc [hunk 23]
```diff
@@ -89,6 +109,9 @@ Required installation configuration parameters are described in the following ta
 |metadata:
   name:
 |The name of the cluster. DNS records for the cluster are all subdomains of `{{.metadata.name}}.{{.baseDomain}}`.
+ifdef::agent[]
+When you do not provide `metadata.name` through either the `install-config.yaml` or `agent-config.yaml` files, for example when you use only ZTP manifests, the cluster name will be set to `agent-cluster`.
```
Comment (kcarmichael08) [inline modules/installation-configuration-parameters.adoc:113]:
```suggestion
When you do not provide `metadata.name` through either the `install-config.yaml` or `agent-config.yaml` files, for example when you use only ZTP manifests, the cluster name is set to `agent-cluster`.
```

File: modules/agent-configuration-parameters.adoc [hunk 24]
```diff
@@ -0,0 +1,123 @@
...
+|Parameter|Description|Values
+
+|rendezvousIP:
+|The IP address of the node that performs the bootstrapping process as well as running the `assisted-service` component.
+You must provide the rendezvous IP address when you do not specify at least one host's IP address in the `networkConfig` parameter.
+If this address is not provided, one IP address is selected from the provided hosts' `networkConfig`.
+|IPv4 or IPv6 address.
+
+|bootArtifactsBaseURL:
+|The URL of the server to upload PXE assets to when using the Agent-based Installer to generate an iPXE script.
```
Comment (kcarmichael08) [inline modules/agent-configuration-parameters.adoc:64]: Not sure if PXE should be spelled out at first use or is a term not normally spelled out.

File: modules/agent-configuration-parameters.adoc [hunk 25]
```diff
@@ -0,0 +1,123 @@
...
+  hostname:
+|Hostname.
+Overrides the hostname obtained from either the Dynamic Host Configuration Protocol (DHCP) or a reverse DNS lookup.
+Each host must have a unique hostname supplied by one of these methods, although configuring a hostname through this parameter is optional.
+|String.
+
+|hosts:
+  interfaces:
+|Provides a table of the name and MAC address mappings for the interfaces on the host.
+If a `NetworkConfig` section is provided in the `agent-config.yaml`, this table must be included and the values must match the mappings provided in the `NetworkConfig` section.
```
Comment (kcarmichael08) [inline modules/agent-configuration-parameters.adoc:88]:
```suggestion
If a `NetworkConfig` section is provided in the `agent-config.yaml` file, this table must be included and the values must match the mappings provided in the `NetworkConfig` section.
```

File: modules/agent-configuration-parameters.adoc [hunk 26]
```diff
@@ -0,0 +1,123 @@
...
+|hosts:
+  interfaces:
+    macAddress:
+|The MAC address of an interface on the host.
+|A MAC address such as the following example: `00-B0-D0-63-C2-26`
+
+|hosts:
+  rootDeviceHints:
+|Enables provisioning of the {op-system-first} image to a particular device.
+It examines the devices in the order it discovers them, and compares the discovered values with the hint values.
```
Comment (kcarmichael08) [inline modules/agent-configuration-parameters.adoc:106]: It = The installation program ?

File: modules/installation-configuration-parameters.adoc [hunk 27]
```diff
@@ -457,6 +485,14 @@ ifdef::ibm-power,ibm-power-vs[]
 |Determines the instruction set architecture of the machines in the pool. Currently, heteregeneous clusters are not supported, so all pools must specify the same architecture. Valid values are `ppc64le` (the default).
 |String
 endif::ibm-power,ibm-power-vs[]
+
+ifdef::agent[]
+|compute:
+  architecture:
+|Determines the instruction set architecture of the machines in the pool. Currently, clusters with varied architectures are not supported. All pools must specify the same architecture. Valid values are `amd64` and `arm64`.
```
Comment (kcarmichael08) [inline modules/installation-configuration-parameters.adoc:492]: Would the "valid values" part go in the values column?

File: modules/installation-configuration-parameters.adoc [hunk 28]
```diff
@@ -537,6 +578,14 @@ ifdef::ibm-power,ibm-power-vs[]
 |Determines the instruction set architecture of the machines in the pool. Currently, heterogeneous clusters are not supported, so all pools must specify the same architecture. Valid values are `ppc64le` (the default).
 |String
 endif::ibm-power,ibm-power-vs[]
+
+ifdef::agent[]
+|controlPlane:
+  architecture:
+|Determines the instruction set architecture of the machines in the pool. Currently, clusters with varied architectures are not supported. All pools must specify the same architecture. Valid values are `amd64` and `arm64`.
```
Comment (kcarmichael08) [inline modules/installation-configuration-parameters.adoc:585]: see previous "values" comment

File: modules/installing-ocp-agent-inputs.adoc [hunk 29]
```diff
@@ -173,7 +173,7 @@ EOF
...
 <2> Optional: Host configuration. The number of hosts defined must not exceed the total number of hosts defined in the `install-config.yaml` file, which is the sum of the values of the `compute.replicas` and `controlPlane.replicas` parameters.
 <3> Optional: Overrides the hostname obtained from either the Dynamic Host Configuration Protocol (DHCP) or a reverse DNS lookup. Each host must have a unique hostname supplied by one of these methods.
-<4> Enables provisioning of the Red Hat Enterprise Linux CoreOS (RHCOS) image to a particular device. It examines the devices in the order it discovers them, and compares the discovered values with the hint values. It uses the first discovered device that matches the hint value.
+<4> Enables provisioning of the {op-system-first} image to a particular device. It examines the devices in the order it discovers them, and compares the discovered values with the hint values. It uses the first discovered device that matches the hint value.
```
Comment (kcarmichael08) [inline modules/installing-ocp-agent-inputs.adoc:176]: It = installation program?

**Review by kcarmichael08 — COMMENTED** [review]
Great job with all the new content! I marked a few minor style things and some questions I had. Hope this helps!

File: installing/installing_with_agent_based_installer/installation-config-parameters-agent.adoc (hunk 17 above)
Comment (skopacz1) [reply installing/installing_with_agent_based_installer/installation-config-parameters-agent.adoc:19]: This is a duplicate of the comment below, so I'm assuming this is meant for the first list item in this Additional Resources list. I'll remove periods from both items.

File: modules/agent-configuration-parameters.adoc (hunk 18 above)
Comment (skopacz1) [reply modules/agent-configuration-parameters.adoc:7]: Good catch, thank you

File: modules/agent-configuration-parameters.adoc (hunk 24 above)
Comment (skopacz1) [reply modules/agent-configuration-parameters.adoc:64]: I'm not sure either, but I can spell it out here anyways

File: installing/installing_with_agent_based_installer/installation-config-parameters-agent.adoc (hunk 17 above)
Comment (kcarmichael08) [reply installing/installing_with_agent_based_installer/installation-config-parameters-agent.adoc:19]: ooops. yes :)

File: modules/installation-configuration-parameters.adoc (hunk 27 above)
Comment (skopacz1) [reply modules/installation-configuration-parameters.adoc:492]: Based on the way this attribute is written for other install pages (see lines 458-487) I guess not. I'd prefer to follow the existing format unless you'd prefer me to put these values in the other column.

File: modules/installation-configuration-parameters.adoc (hunk 28 above)
Comment (skopacz1) [reply modules/installation-configuration-parameters.adoc:585]: Based on the way this attribute is written for other install pages (see lines 551-580) I guess not. I'd prefer to follow the existing format unless you'd prefer me to put these values in the other column.

File: modules/installing-ocp-agent-inputs.adoc (hunk 29 above)
Comment (skopacz1) [reply modules/installing-ocp-agent-inputs.adoc:176]: I think so, yeah.
